sklearn/utils/extmath.py

```diff
@@ -1128,11 +1132,11 @@ def _incremental_mean_and_var(
             # equivalent to np.nansum((X-T)**2 * sample_weight, axis=0)
             # safer because np.float64(X*W) != np.float64(X)*np.float64(W)
             correction = _safe_accumulator_op(
-                np.matmul, sample_weight, np.where(X_nan_mask, 0, temp)
+                xp.matmul, sample_weight, np.where(X_nan_mask, 0, temp)
             )
             temp **= 2
             new_unnormalized_variance = _safe_accumulator_op(
-                np.matmul, sample_weight, np.where(X_nan_mask, 0, temp)
+                xp.matmul, sample_weight, np.where(X_nan_mask, 0, temp)
```

Comment:
Why is np.where used here?

Reply:
I think this is to avoid applying the `temp` correction on entries that correspond to nan values in a row of `X`.